lib/galaxy/objectstore/s3.py

```diff
@@ -23,14 +23,18 @@
 
 from .s3_multipart_upload import multipart_upload
 from ..objectstore import convert_bytes, ObjectStore
+from cloudbridge.cloud.factory import CloudProviderFactory, ProviderList
 
+# TODO: Cloudbridge is not exposing exceptions; however, it is planned in the next release milestone.
```

Comment:
Few `todos` are removed. IMHO the ones that are left would make it easier for future development. But if it's still those are too many, I can remove less important ones.